### ISSUE TITLE
3d

### ISSUE THREAD

**Issue by gedw99** [hemanthrajv/flutter_3d_obj#1]
I do the same sort of 3d with flutter.

I have been looking into rendering a 3d view that can be panned and zoomed and orbit around. Even doing picking.

Now flutter does not have a 3d view exposed but it is skia under the hood.
So there is hope.

One idea is to build a 3d engine as a plug-in that does have opengl / metal engine. This gets you the speed.
Then using Method Channels it's possible to output the rendered images to flutter at very high frame rate. It's called asynchonous resource loading in flutter. It's how they can make a scrolling list do fast 

Wondering if you have looked into this ?

**Comment by hemanthrajv** [comment]
Hi,
Thanks for your views.I wrote the library way long back when flutter had no
way to show native views. The flutter team however has idea to provide a
full 3d API in near future. Your idea seems worth trying,I'll for sure give
it a try.

Thanks,
Hemanth Raj

On 09-Mar-2018 5:48 PM, "gedw99" <notifications@github.com> wrote:

I do the same sort of 3d with flutter.

I have been looking into rendering a 3d view that can be panned and zoomed
and orbit around. Even doing picking.

Now flutter does not have a 3d view exposed but it is skia under the hood.
So there is hope.

One idea is to build a 3d engine as a plug-in that does have opengl / metal
engine. This gets you the speed.
Then using Method Channels it's possible to output the rendered images to
flutter at very high frame rate. It's called asynchonous resource loading
in flutter. It's how they can make a scrolling list do fast

Wondering if you have looked into this ?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/hemanthrajv/flutter_3d_obj/issues/1>, or mute the thread

**Comment by gedw99** [comment]
Thanks for quick response.

i read on their web site that they do "plan" to do a 3D API one day.

Funny thing is that Microsoft XAMl engine uses Skia under the hood. They
call its SkiaSharp of course . Its quite hilarious - Microsoft's key big
name product uses Skia for most of the work !
Any they support all desktops and mobiles.
What a funny war is front of our eyes :)


https://docs.microsoft.com/en-us/windows/uwp/design/layout/3-d-perspective-effects

https://docs.microsoft.com/en-us/uwp/api/Windows.UI.Xaml.Media.Media3D.Matrix3D

---

So i was thinking about using golang, because i already use it with Flutter
and Servers and well basically everything.
I can bind golang to xaml and flutter now. Its not easy with xaml but works.
But its a bit difficult because neither quite let you get into the right
layer of the Architecture.
SO i am playing aroudn with the idea of Materialized Views that basically
vector data structures and such things that each library can render in 3D.
This way the logic and complexity stays at the golang level and all the
easy many CPU scaling.
And the materialised Views allow you to have a PUB SUB architecture with
mutations coming in from the GUI Layer, golang system doing its work, and
outputting mutations of the changed Vectors back to the GUi Layer. Its
basically like GraphQL come to think of it.

Its a bit of a mad science experiment i agree..

But GraphQL is a pretty good abstraction for this type of problem.

If you feel like commenting if you have ideas in this area go for it.
But i do understand this is really off the deep end a it.

**Comment by hemanthrajv** [comment]
I don't have pretty much knowledge about golang, but your idea seems quite
interesting. Would like to see it action. 😊

**Comment by gedw99** [comment]
Will investigate it..
